.github/workflows/deploy.yml

```diff
@@ -124,8 +127,11 @@ jobs:
         timeoutSeconds: 300
         intervalSeconds: 15
 
+    - name: Exit whole workflow if smoke test was not successful
+      if: steps.wait_for_smoke_test.outputs.conclusion != 'success'
+      run: exit 1
+
     - name: Trigger Deploy App Workflow for production
-      if: ${{ steps.wait_for_smoke_test.outputs.conclusion == 'success' }} && github.ref == 'refs/heads/master'
```

Comment:
is it worth having `github.ref == 'refs/heads/master'` 
it could protect from accidentally deploying to production if you were to test pipeline changes from a different branch?

Reply:
Thanks. I've reinstated the branch checks now that I've proved the boolean logic without curly braces.